src/uu/paste/src/paste.rs

```diff
@@ -89,6 +90,53 @@ pub fn uu_app() -> Command {
         )
 }
 
+struct DelimiterData<'a> {
+    current_delimiter_length: usize,
+    delimiters_encoded: &'a [Box<[u8]>],
+    delimiters_encoded_iter: Iter<'a, Box<[u8]>>,
+}
+
+/// - If there are no delimiters, returns `None`
+/// - If there are delimiters, tries to return the next delimiter
+/// - If the end of the delimiter list was reached, resets the iter to point to the beginning of the delimiter list
+///     - (Technically this is done by creating a new iter)
```

Comment:
Just a thought: did you consider to use an infinite iterator with `cycle`?

Reply:
That's cool, I was thinking something like that should exist. Updated the PR to use `cycle`.